test/test_api/test_project_api.py

```diff
@@ -1154,6 +1154,38 @@ def test_user_progress_guidelines_updated(self):
         assert data['guidelines_updated'] is True, data
 
 
+    @with_context
+    def test_task_progress(self):
+        """Test API taskprogress as anonymous works"""
+        from pybossa import data_access
+        user = UserFactory.create(info=dict(data_access=['L1']))
+        project = ProjectFactory.create(owner=user)
+        tasks = TaskFactory.create_batch(2, project=project)
+        headers = [('Authorization', user.api_key)]
+        category = CategoryFactory.create()
+        res = self.app.get('/api/project?all=1&category_id=%s' % category.id, headers=headers)
+
+        taskruns = []
+        for task in tasks:
+            taskruns.extend(AnonymousTaskRunFactory.create_batch(2, task=task))
+
+        # check basic query without constraints to filter tasks  
+        res = self.app.get('/api/project/1/taskprogress', follow_redirects=True)
```

Comment:
```
res = self.app.get
(url = '/api/project/1/taskprogress?api_key=%s' % user.api_key)
```

Reply:
Thanks, using the header with the request worked for authorization!